src/quemb/molbe/opt.py

```diff
@@ -174,24 +177,34 @@ def optimize(self, method, J0=None, trust_region=False):
                 self.objfunc, array(self.pot), f0, J0, max_space=self.max_space
             )
 
+            if settings.PRINT_LEVEL >= 10:
+                print(step0_timer.str_elapsed())
             if self.err < self.conv_tol:
                 print(flush=True)
                 print("CONVERGED w/o Optimization Steps", flush=True)
                 print(flush=True)
             else:
                 # Perform optimization steps
                 for iter_ in range(self.max_space):
+                    iter_timer = Timer("Time to complete Iteration " + str(self.iter))
                     print("-- In iter ", self.iter, flush=True)
                     optQN.next_step(trust_region=trust_region)
                     self.iter += 1
                     print(
                         "Error in density matching      :   {:>2.4e}".format(self.err),
                         flush=True,
                     )
+                    if settings.PRINT_LEVEL >= 10:
+                        print(iter_timer.str_elapsed())
                     print(flush=True)
                     if self.err < self.conv_tol:
                         print(flush=True)
                         print("CONVERGED", flush=True)
+                        if settings.PRINT_LEVEL >= 10:
```

Comment:
If you want to change the message later it is probably better to do:
```python
def str_elapsed(message: str | None = None) -> str:
    return f"{self.message if message is None else message}: {self.elapsed():.5f}"
```

and do
```python3
print(step0_timer.str_elapsed("Total time to complete BE optimization"))
```

I will approve anyway, so you can do whatever you prefer.

Reply:
cool thanks! this is helpful. i'll reflect those changes and merge :)